### ISSUE TITLE
clasp push --watch will check manifest change per each changed file

### ISSUE THREAD

**Issue by OleksandrDemian** [google/clasp#905]
## Expected Behavior
When updating multiple files at once in push watch mode (ex: I modify 5 files and then hit "Save all") I would expect only 1 push and 1 manifest changes check.

## Actual Behavior
When updating multiple files at once in push watch mode, you will get 1 push (because it is debounced), but as many `manifestHasChanges` requests as there are updated files.

This happens because `manifestHasChanges` is not debounced, and as conseguence you call it as many times as there are changed files (`manifestHasChanges` will fetch AppsScript project and check if manifest changed).

## Steps to Reproduce the Problem
1. Create an AppsScript project
2. Create a couple of files (2-3 is enought)
3. Modify the files with your editor without saving them one by one
4. Save all modified files
5. See multiple requests starting for `manifestHasChanges` (you can simply put a logline there).

## Specifications
- Node version 10.17.6:
- Version 2.4.1:
- OS Windows:

A possible fix can be as simple as moving the manifest file check into the debounce function.

Current implementation:
```js
if (options.watch) {
        console.log(LOG.PUSH_WATCH);
        // Debounce calls to push to coalesce 'save all' actions from editors
        const debouncedPushFiles = debounce(() => {
            console.log(LOG.PUSHING);
            return pushFiles();
        }, WATCH_DEBOUNCE_MS);
        const patterns = await DOTFILE.IGNORE();
        const watchCallback = async (filePath) => {
            if (multimatch([filePath], patterns, { dot: true }).length > 0) {
                // The file matches the ignored files patterns so we do nothing
                return;
            }
            console.log(`\n${LOG.PUSH_WATCH_UPDATED(filePath)}\n`);
            if (!options.force && (await manifestHasChanges(projectSettings)) && !(await confirmManifestUpdate())) {
                console.log('Stopping push…');
                return;
            }
            return debouncedPushFiles();
        };
        const watcher = chokidar.watch(rootDir, { persistent: true, ignoreInitial: true });
        watcher.on('ready', pushFiles); // Push on start
        watcher.on('add', watchCallback);
        watcher.on('change', watchCallback);
        watcher.on('unlink', watchCallback);
        return;
    }
```

Working solution:
```js
if (options.watch) {
        console.log(LOG.PUSH_WATCH);
        // Debounce calls to push to coalesce 'save all' actions from editors
        const debouncedPushFiles = debounce(async () => {
            if (!options.force && (await manifestHasChanges(projectSettings)) && !(await confirmManifestUpdate())) {
                console.log('Stopping push…');
                return;
            }

            console.log(LOG.PUSHING);
            return pushFiles();
        }, WATCH_DEBOUNCE_MS);
        const patterns = await DOTFILE.IGNORE();
        const watchCallback = async (filePath) => {
            if (multimatch([filePath], patterns, { dot: true }).length > 0) {
                // The file matches the ignored files patterns so we do nothing
                return;
            }
            console.log(`\n${LOG.PUSH_WATCH_UPDATED(filePath)}\n`);
            return debouncedPushFiles();
        };
        const watcher = chokidar.watch(rootDir, { persistent: true, ignoreInitial: true });
        watcher.on('ready', pushFiles); // Push on start
        watcher.on('add', watchCallback);
        watcher.on('change', watchCallback);
        watcher.on('unlink', watchCallback);
        return;
    }
```

**Comment by HoldYourWaffle** [comment]
Should this be closed now that #919 has been merged?

**Comment by OleksandrDemian** [comment]
Indeed
